ordinary_data/valve/tr_valve_pipe.sql

```diff
@@ -5,7 +5,49 @@
 */
 
 /* UPDATES NUMBER OF VALVES AND CLOSED/OPEN FOR PIPES */
-CREATE OR REPLACE FUNCTION qwat_od.fn_pipe_update_valve(_pipe_id integer) RETURNS VOID AS
+CREATE OR REPLACE FUNCTION qwat_od.fn_pipe_delete_valve(_valve_id integer)
+  RETURNS void AS
+$BODY$
+BEGIN
+
+IF (SELECT _valve_count > 1 FROM qwat_od.pipe WHERE id = (
+					SELECT fk_pipe
+					FROM qwat_od.valve
+					WHERE id = _valve_id))
+THEN
+			UPDATE qwat_od.pipe
+			SET
+				_valve_count  = valve_group.vcount,
+				_valve_closed = valve_group.vclosed
```

Comment:
can you detail the logic?
it's not possible to handle this in an AFTER trigger?

Reply:
It's the same as update instead of I excude the valve_id from the count (for bool_.

Reply:
yep, I got the idea, but it seems a bit fragile to me (there is no control that the -1 actually corresponds to anything)
That's why I'm asking if we can't have a simpler function in an after trigger (hence no need for the -1).

Reply:
Done